pkg/operator/ceph/client/controller.go

```diff
@@ -375,31 +414,41 @@ func generateClientName(name string) string {
 }
 
 // updateStatus updates an object with a given status
-func (r *ReconcileCephClient) updateStatus(observedGeneration int64, name types.NamespacedName, status cephv1.ConditionType) {
-	cephClient := &cephv1.CephClient{}
-	if err := r.client.Get(r.opManagerContext, name, cephClient); err != nil {
-		if kerrors.IsNotFound(err) {
-			logger.Debug("CephClient resource not found. Ignoring since object must be deleted.")
-			return
+func (r *ReconcileCephClient) updateStatus(observedGeneration int64, name types.NamespacedName, status cephv1.ConditionType, cephx *cephv1.CephxStatus) {
+	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		cephClient := &cephv1.CephClient{}
+		if err := r.client.Get(r.opManagerContext, name, cephClient); err != nil {
+			if kerrors.IsNotFound(err) {
+				logger.Debug("CephClient resource not found. Ignoring since object must be deleted.")
+				return nil
+			}
+			logger.Warningf("failed to retrieve ceph client %q to update status to %q. %v", name, status, err)
+			return errors.Wrapf(err, "failed to retrieve ceph client %q to update status to %q", name, status)
+		}
+		if cephClient.Status == nil {
+			cephClient.Status = &cephv1.CephClientStatus{}
 		}
-		logger.Warningf("failed to retrieve ceph client %q to update status to %q. %v", name, status, err)
-		return
-	}
-	if cephClient.Status == nil {
-		cephClient.Status = &cephv1.CephClientStatus{}
-	}
 
-	cephClient.Status.Phase = status
-	if cephClient.Status.Phase == cephv1.ConditionReady {
-		cephClient.Status.Info = generateStatusInfo(cephClient)
-	}
-	if observedGeneration != k8sutil.ObservedGenerationNotAvailable {
-		cephClient.Status.ObservedGeneration = observedGeneration
-	}
-	if err := reporting.UpdateStatus(r.client, cephClient); err != nil {
-		logger.Errorf("failed to set ceph client %q status to %q. %v", name, status, err)
-		return
+		cephClient.Status.Phase = status
+		if cephClient.Status.Phase == cephv1.ConditionReady {
+			cephClient.Status.Info = generateStatusInfo(cephClient)
+		}
+		if observedGeneration != k8sutil.ObservedGenerationNotAvailable {
+			cephClient.Status.ObservedGeneration = observedGeneration
+		}
+		if cephx != nil {
+			cephClient.Status.Cephx = *cephx
+		}
+		if err := reporting.UpdateStatus(r.client, cephClient); err != nil {
+			logger.Errorf("failed to set ceph client %q status to %q. %v", name, status, err)
+			return errors.Wrapf(err, "failed to set ceph client %q status to %q", name, status)
+		}
+		return nil
+	})
+	if err != nil {
+		logger.Error(err)
```

Comment:
Santosh discovered a bug earlier in RGW's update implementation. This should return an error.

Reply:
@sp98  would have fixed dont procceed if the key is not rotated https://github.com/rook/rook/pull/16091

In this check we are retrying to update the status, currently in the rook base we never return err on failed update of status,
1) If needed I can include the fail err for cephclient with this PR

2) Open a issue, where we can discuss on how to handle status update for all the controllers

Reply:
Please fix the issue here. Because status updates are critical for this feature, status update checks are a ripple effect requirement introduced by this work.

Reply:
Updated